client/knode/validator/states.go

```diff
@@ -29,44 +29,10 @@ type work struct {
 
 type stateFn func() stateFn
 
-func (val *validator) notLoggedInState() stateFn {
-	isGenesis, err := val.consensus.IsGenesisValidator(val.walletAccount.Account().Address)
-	if err != nil {
-		log.Warn("Failed to verify the voter information", "err", err)
-		return nil
-	}
-
-	// @NOTE (rgeraldes) - sync was already done at this point and by default the investors will be
-	// part of the initial set of validators - no need to make a deposit if the block number is 0
+func (val *validator) genesisNotLoggedInState() stateFn {
+	// no need to make a deposit if the block number is 0
 	// since these validators will be marked as voters from the start
-	if !isGenesis || (isGenesis && val.chain.CurrentBlock().NumberU64() > 0) {
-		chainHeadCh := make(chan core.ChainHeadEvent)
-		chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
-		defer chainHeadSub.Unsubscribe()
-
-		isValidator, err := val.consensus.IsValidator(val.walletAccount.Account().Address)
-		if err != nil {
-			log.Crit("Failed to verify if account is already a validator")
-		}
-
-		if !isValidator {
-			txHash, err := val.consensus.Join(val.walletAccount, val.deposit)
-			if err != nil {
-				log.Error("Error joining validators network", "err", err)
-				return nil
-			}
-			log.Info("Waiting confirmation to participate in the consensus")
-
-			receipt, err := tx.WaitMinedWithTimeout(val.backend, txHash, txConfirmationTimeout)
-			if err != nil {
-				log.Crit("Failed to verify the voter registration", "err", err)
-			}
-			if receipt.Status == types.ReceiptStatusFailed {
-				log.Crit("Failed to register the validator - receipt status failed")
-			}
-		}
-
-	} else {
+	if val.isBlockZero() {
 		isVoter, err := val.consensus.IsValidator(val.walletAccount.Account().Address)
```

Comment:
This check is not necessary as according to the contract, in order to be a genesis on block 0, the user needs to be registered as a validator.

Reply:
not sure what you mean with remove the check.

Does that means that a genesis validator never does a deposit?

Reply:
Oh I see, got the line worng, so as long is a genesis and block zero can just start validating imediatly

Reply:
yes